teamMaster.py

```diff
@@ -0,0 +1,264 @@
+        yyyy = gamedate[6:10]
+        mm = gamedate[0:2]
+        dd = gamedate[3:5]
+    else:
+        yyyy = datetime.date.today().strftime("%Y")
+        mm = datetime.date.today().strftime("%m")
+        dd = datetime.date.today().strftime("%d")
+
+    mmddyyyy = mm + '-' + dd + '-' + yyyy
+    schedule_input = DAILY_SCHEDULE.replace('YYYYMMDD', yyyy + mm + dd)
+    team_output = TEAM_MSTR_O.replace('YYYYMMDD', yyyy + mm + dd)
+
+    logger.info('dailySched dictionary location: ' + schedule_input)
+    logger.info('Updated team dictionary location: ' + team_output)
+
+    return (schedule_input, team_output, mmddyyyy)
+
+
+def load_daily_schedule(schedule_in):
+    """
+    :param schedule_in: filename for daily schedule file to load into memory
+    """
+
+    logger.info('Loading dailySchedule dictionary')
+
+    try:
+        with open(schedule_in) as schedulefile:
+            return json.load(schedulefile)
+    except Exception as e:
+        errmsg = 'Error loading dailySchedule dictionary. . .'
+        logger.critical(errmsg)
+        logger.exception(e)
+        raise LoadDictionaryError(errmsg)
+
+
+def reset_todays_games_in_master(masterdict):
+
+    keylist = list(masterdict)
+    for key in keylist:
+        keys = list(masterdict[key])
+        if 'today_1' in keys:
+            del masterdict[key]['today_1']
+        if 'today_2' in keys:
+            del masterdict[key]['today_2']
+
+    return masterdict
+
+
+def process_schedule(sched_dict, mstr_dict):
+    """
+    :param sched_dict: daily schedule file used to drive team extract process
+    """
+
+    mstr_dict = reset_todays_games_in_master(mstr_dict)
+
+    logger.info('Applying updates to team master dictionary')
+
+    # get dailySchedule key list and start with blank result team dictionary
+    keylist = list(sched_dict)
+    daily_team_dict = {}
+
+    # for each key, go to boxscore for that game and exctract all team data
+    for key in keylist:
+
+        boxscoreurl = BOXSCORE.replace('/_directory_/',
+                                       sched_dict[key]['directory'])
+        logger.info('Loading boxscore dictionary: ' + boxscoreurl)
+
+        try:
+            boxresp = requests.get(boxscoreurl)
+            boxscore_dict = json.loads(boxresp.text)
+            box_level_3 = boxscore_dict['data']['boxscore']
+        except Exception:
+            errmsg = 'Boxscore dictionary not created yet for: ' + key
+            logger.warning(errmsg)
+            continue
+
+        # set home and away team codess, init team result dict for this game
+        home = sched_dict[key]['home_code']
+        away = sched_dict[key]['away_code']
+
+        # single game or first game of a double header
+        if key[-1:] == '1':
+            try:
+                hist = mstr_dict[home]['game_hist']
+                hist.update({box_level_3['game_id']:
+                             sched_dict[key]['directory']})
+            except Exception:
```

Comment:
There is a comment both if the try works and if the exception is raised...

    try:
        # grab team game history from master and add current game
        hist = mstr_dict[team]['game_hist']
        hist.update({box_level_3['game_id']: game_dir})
    except Exception:
        # first time game history being created for this team
        hist = {box_level_3['game_id']: game_dir}

Does it need to be more detailed? Basically if there is no game history for the team, meaning today is their first game, then establish the 'game_hist' subdictionary within the team entry using the current game.